include/deal.II/grid/grid_tools.h

```diff
@@ -356,6 +357,77 @@ namespace GridTools
                        Triangulation<dim, spacedim> &triangulation,
                        const bool   keep_boundary=true);
 
+  /**
+   * Remove hanging nodes from a grid. If the @p isotropic parameter is set
+   * to @p false (default) this function detects cells with hanging nodes and
+   * refines the neighbours in the direction that removes hanging nodes.
+   * If the @p isotropic parameter is set
+   * to @p true, the neighbours refinement is made in each directions.
+   * In order to remove all hanging nodes this procedure has to be repeated:
+   * this could require a large number of iterations.
+   * To avoid this a max number (@p max_iterations) of iteration is provided.
+   *
+   * Consider the following grid:
+   * @image html remove_hanging_nodes-hanging.png
+   *
+   * @p isotropic == @p false would return:
+   * @image html remove_hanging_nodes-aniso.png
+   *
+   * @p isotropic == @p true would return:
+   * @image html remove_hanging_nodes-isotro.png
+   *
+   * @param[in,out] tria Triangulation to refine.
+   *
+   * @param[in] isotropic if true refine cells in each directions, otherwise
+   * (default value) refine the cell in the direction that removes hanging node.
+   *
+   * @param[in] max_iterations at each step only closest cells to hanging nodes
+   * are refined. The code may require a lot of iterations to to remove all
+   * hangind nodes. @p max_iterations is the maximum number of iteration
+   * allowed. If @p max_iterations == 0 this function continues refining until
```

Comment:
In essence, you are designating zero to be a special, "invalid" value for the number of iterations. We typically use `numbers::invalid_unsigned_int` for this. I would prefer if that was the case.